File: src/main/java/com/staymate/service/RoomService.java
package com.staymate.service;

import java.util.List;

import com.staymate.dto.RoomDTO;
import com.staymate.entity.Room;

public interface RoomService {

    Room addRoom(RoomDTO dto);

    List<Room> getRooms(String hostelCode);

    Room updateRoom(Long id, RoomDTO dto);

    void deleteRoom(Long id);

}

File: frontend/src/services/roomService.js
import api from "./api";

const getRooms = (hostelCode) =>
  api.get(`/rooms/hostel/${hostelCode}`);

const getRoom = (id) =>
  api.get(`/rooms/${id}`);

const addRoom = (room) =>
  api.post("/rooms", room);

const updateRoom = (id, room) =>
  api.put(`/rooms/${id}`, room);

const deleteRoom = (id) =>
  api.delete(`/rooms/${id}`);

export default {
  getRooms,
  getRoom,
  addRoom,
  updateRoom,
  deleteRoom,
};